enhancements/accelerators/out-of-tree-drivers.md

```diff
@@ -0,0 +1,418 @@
+      paths:
+      - sourcePath: "/usr/src/<vendor>/<artifact>
+        destinationDir: "/usr/src/"
+      claims:
+      - name: "<vendor>-pvc"
+      mountPath: "/usr/src/<vendor>-<internal>"
+  node: 
+    selector: "feature.../pci-<VENDOR_ID>.present"
+  dependsOn:
+    - name: <CR_NAME_VENDOR_ID_XYZ>
+    - name: <CR_NAME_VENDOR_ID_KJI>
+```
+
+With the above information XYZ is capable of deducing all needed information to build and manage a DriverContainer. All manifests in XYZ are templates that are templatized during reconciliation with runtime and meta information. 
+
+The name e.g. is used to prefix all resources (Pod, DameonSet, RBAC, ServiceAccount, Namespace, etc) created for this very specific **{vendor}-{hardware}**. The DriverContainer section expects at least the git repository (mandatory) from a vendor. This repository has all tools and scripts to build the kernel module. The base image for a DriverContainer is an UBI7,8  with the KVC (kmod-via-containers)  framework installed. 
+
+KVC provides hooks to build, load, unload the kernel modules and a wrapper for userspace utilities. We might extend the number of hooks to have a similar interface as dkms. 
+Another important field is the build arguments. We have often seen incompatibility between workloads and driver versions. Selecting a specific version is sometimes the only way to have a workload successfully running on OpenShift or BareMetal.  This field can also be used by an administrator to upgrade or downgrade a kernel module due to CVEs, bug fixes or incompatibility. Some drivers have also some flags  to enable or disable specific features of a driver. 
+
+The next section is used to filter the nodes on which a kernel module or driver should be deployed on. It makes no sense to deploy drivers on nodes where the hardware is not available. Furthermore this can also be used to target even subsets of special nodes either by creating labels manually or leveraging NFDs hook functionality. 
+
+The next section is used to tell XYZ where to find build artifacts from other drivers. Some drivers need e.g. symbol information from kernel modules, header files or the complete driver sources to be built successfully.  We are providing two ways for these artifacts to be consumed.  (1) Some vendors expose the build artifacts in a hostPath. The DriverContainer with KVC needs a hook for preparing the sources, which means it would copy from sourcePath on the host to the destinationDir in the DriverContainer. (2) The other way to get build artifacts is to use an DriverContainer image that is already built do get the needed artifacts (We are assuming here that the vendor is not exposing any artifacts to the host). We can leverage those images in a multi-stage build for the DriverContainer. 
+
+To retrieve the correct image we are using XYZs templating to inject the correct runtime information, here we are using **{{.KernelVersion}}** as a unique identifier for DriverContainer images. 
```

Comment:
https://github.com/openshift/enhancements/blob/master/enhancements/rhcos/extensions.md may be of interest here as well as `machine-os-content`'s will be providing kernel-devel/headers to be inline with what the nodes should be running. While the extension system itself may not be key for this, the available packages may be helpful.